### ISSUE TITLE
Symlink files not copied to tmp dir for Integration tests

### ISSUE THREAD

**Issue by phewitt** [pulumi/pulumi#3735]
It seems that when I have a symlinked file inside of the pulumi stack at test. When I run `integration.ProgramTest` all files are copied to `tmp` besides symlinks. Haven't had time to test in isolation. Simply have experienced this in one of our projects.

**Comment by lukehoban** [comment]
@phewitt For your use-case, what behaviour did you expect/need for symlinks?  Did you want them to be followed and copied as real files, or remain as links to the source location?  Neither option seems generally "safe" here (the former could lead to cycles, the latter could lead to accidentally have the tests not perform isolated operations.

**Comment by lukehoban** [comment]
FWIW - the issue appears to be we use Go's `fsutil.CopyFile` which includes this documentation:

```go
// CopyFile is a braindead simple function that copies a src file to a dst file.  Note that it is not general purpose:
// it doesn't handle symbolic links, it doesn't try to be efficient, it doesn't handle copies where src and dst overlap,
// and it makes no attempt to preserve file permissions.  It is what we need for this utility package, no more, no less.
```

**Comment by phewitt** [comment]
for our use case we originally were we had a directory `/docker` which had many sub directories for each unique image `/docker/image1/` `/docker/image2/`. However we had common bash scripts that we were using in each container, so we moved it out of copy pasting into each `image` directory and made it just `/docker/common.sh` (didn't actually name it common). and then symlink inside of each `/docker/image` directory. Then we use a ConfigMap `/docker/image1/common.sh` to mount the sh file into each container. However, I realized that it is simpler to just mount the same config map instead of symlinking. IE all containers use 1 configmap using `/docker/common.sh` instead of making many config maps.

So, for us we don't actually need it now! But, i figured it might make sense to look at for others use case. After you explain the downsides of copying as real files or remain as a link, I don't know what would make sense! thanks for taking the time.
